plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,300 @@
 
-      nixGrammars = mkOption {
-        type = types.bool;
-        default = true;
-        description = "Install grammars with Nix";
-      };
+  maintainers = [ ];
 
-      ensureInstalled = mkOption {
-        type =
-          with types;
-          oneOf [
-            (enum [ "all" ])
-            (listOf str)
-          ];
-        default = "all";
-        description = "Either \"all\" or a list of languages";
-      };
+  # TODO introduced 2024-07-06: remove after 24.11
+  optionsRenamedToSettings = [
+    "ensureInstalled"
+    "ignoreInstall"
+    "indent"
+    "parserInstallDir"
+    [
+      "incrementalSelection"
+      "enable"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeDecremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeIncremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "scopeIncremental"
+    ]
+  ];
 
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "customCaptures" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "custom_captures"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "disabledLanguages" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "disable"
+        ]
+      ))
+    ];
 
-      parserInstallDir = mkOption {
-        type = types.nullOr types.str;
-        default = if cfg.nixGrammars then null else "$XDG_DATA_HOME/nvim/treesitter";
-        description = ''
-          Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
-          This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
-        '';
-      };
+  settingsOptions = {
+    additional_vim_regex_highlighting =
+      with helpers.nixvimTypes;
+      helpers.defaultNullOpts.mkNullableWithRaw (either bool (listOf (maybeRaw str))) false ''
+        Setting this to true will run `:h syntax` and tree-sitter at the same time.
+        Set this to `true` if you depend on 'syntax' being enabled (like for indentation).
+        Using this option may slow down your editor, and you may see some duplicate highlights.
+        Instead of true it can also be a list of languages
```

Comment:
Not saying this is wrong, but just checking if you're aware of how commonmark markdown handles linebreaks?

TL;DR, a fully blank line is needed for a paragraph break.\
Or you can render a `<br>` by ending a line with `\`.

Reply:
I think I just copied and pasted this from their documentation. I can force the linebreaks to increase readability.